### ISSUE TITLE
net/haproxy: Weak ciphers

### ISSUE THREAD

**Issue by StrikerTwo** [opnsense/plugins#375]
ssllabs.com gives my websites (IIS sites running behind HAProxy with Let's Encrypt certs) only a "B" rating because:

> This server supports weak Diffie-Hellman (DH) key exchange parameters.

> This server accepts RC4 cipher, but only with older protocols. Grade capped to B.

Anything I can do about this?

I found this: https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/ and this: https://mozilla.github.io/server-side-tls/ssl-config-generator/ - can I do this in the OPNsense GUI? Shouldn't it be the default to disable weak ciphers nowadays?

**Comment by fraenki** [comment]
@StrikerTwo Thanks for the report! I'll make these parameters configurable.

**Comment by fraenki** [comment]
> Can I do this in the OPNsense GUI?

_Technically_ this was already possible in advanced mode (hidden setting), but I understand that this was not user-friendly at all. I've introduced several new settings to make it easier:

![screen_frontend](https://user-images.githubusercontent.com/909706/32974313-58f8c950-cbfd-11e7-965b-feda0bcb9bb6.png)
(Frontend aka "Public Service" configuration)

Some settings can also be configured as global defaults:

![screen_defaults](https://user-images.githubusercontent.com/909706/32974341-82f9ef54-cbfd-11e7-8df5-487fe7dfae8a.png)

You still need to know the cipher list, but now it's a simple copy'n'paste.

> Shouldn't it be the default to disable weak ciphers nowadays?

You're right and I've added new default values. However, I'm not enforcing these values as it would risk breaking existing setups. You'll have to enable the **advanced SSL settings** (see screenshot).
